src/sst/core/link.cc

```diff
@@ -87,6 +100,10 @@ Link::prepareForComplete()
     if ( POLL == type ) { delete pair_link->send_queue; }
 
     pair_link->send_queue = nullptr;
+
+    // If my pair link is a SYNC link, also need to call
+    // finalizeConfiguration on it
```

Comment:
prepareForComplete

Reply:
I have another PR that's waiting on this one to merge so I can rebase.  If you don't find anything substantial in this PR, I've made the edit in the new one.